batch.py

```diff
@@ -216,17 +509,189 @@ def make_gene_loc_dict(file) -> dict[str, dict]:
     return gene_dict
 
 
-def remove_sc_outliers(df, outliers=['966_967', '88_88']):
+# can probably be merged with below
+def extract_genes(gene_list, expression_tsv_path) -> list[str]:
+    """
+    Takes a list of all genes and subsets to only those
+    present in the expression file of interest
+    """
+    expression_df = pd.read_csv(AnyPath(expression_tsv_path), sep='\t')
+    expression_df = filter_lowly_expressed_genes(expression_df)
+    gene_ids = set(list(expression_df.columns.values)[1:])
+    genes = set(gene_list).intersection(gene_ids)
+
+    print("Extracting genes")
+    print(gene_list)
+    print(gene_ids)
+    print(list(sorted(genes)))
```

Comment:
Should these persist, or could they change to use the logging library + include extra debug information if required.